### DISCUSSION TITLE
ProfitOrLossReporter and creating reports for battery

### DISCUSSION THREAD

**Discussion by tnlrgn2** [SeitaBV/timely-beliefs#183] [Q&A] [answered]
Hi;

Reporting has nice tutorials which adds a process and creates different costs and reports ProfitOrLoss.

I wonder if reporter can also create a report for profit or loss for a battery (which can charge and discharge). 
I have tried to create a report for the battery I created with Tutorial I but always got 0 euros at Breakable report even thought system says:
 flexmeasures add report --source 4    --parameters breakable-parameters.json    --start-offset DB,-1D --end-offset DB,2D

Report scope:
        start: 2024-06-09 00:00:00+02:00
        end:   2024-06-12 00:00:00+02:00
Reporter `ProfitOrLossReporter` fetched successfully from the database.
Report computation is running...
Report computation done for sensor `costs (Breakable)`.
Saving report for sensor `costs (Breakable)` to the database...
Success. The report for sensor `costs (Breakable)` has been saved to the database.
Success.

where reporter is:

type: reporter
========
  ID  Name          User ID    Model                 Version    Attributes
----  ------------  ---------  --------------------  ---------  ------------------------------------------
   4  FlexMeasures             ProfitOrLossReporter             {
                                                                    "data_generator": {
                                                                        "config": {
                                                                            "consumption_price_sensor": 1,
                                                                            "loss_is_positive": true

and 
flexmeasures add toy-account --kind reporter

The sensor recording day-ahead prices is day-ahead prices (ID: 1).
Created grid connection capacity
The sensor storing the grid connection capacity of the building is grid connection capacity (ID: 9).
Created headroom
The sensor storing the headroom is headroom (ID: 10).
Created <GenericAsset None: 'toy-process' (process)>
Created costs (Inflexible)
The sensor storing the loss is costs (Inflexible) (ID: 11).
Created costs (Breakable)
The sensor storing the loss is costs (Breakable) (ID: 12).
Created costs (Shiftable)
The sensor storing the loss is costs (Shiftable) (ID: 13).
[FLEXMEASURES][2024-06-10 20:04:54,930] INFO: Setting up FlexMeasures's ProfitOrLossReporter model as new data source...
Reporter `ProfitOrLossReporter` saved with the day ahead price sensor in the `DataSource` (id=4)

**Comment by Flix6x** [comment, marked as answer]
Your question is not related to the `timely-beliefs` package, but to `flexmeasures`. Unfortunately, I cannot transfer this discussion because these two packages live under different organisations (even though our company is maintaining both). [Please repost here](https://github.com/FlexMeasures/flexmeasures/discussions/new?category=q-a).
